synchronicity/synchronizer.py

```diff
@@ -44,6 +44,20 @@ def run_forever():
         is_ready.wait()  # TODO: this might block for a very short time
         return self._loop
 
+    def _close_loop(self):
+        if self._loop is not None:
+            self._loop.call_soon_threadsafe(self._loop.stop)
+            while self._loop.is_running():
```

Comment:
isn't there a way to wait for the loop to finish

Reply:
couldn't find a way to do that unfortunately